jpsShared/src/org/jetbrains/jps/incremental/scala/model/HydraSettingsImpl.java

```diff
@@ -4,34 +4,44 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.jps.model.ex.JpsElementBase;
 
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
 /**
  * @author Maris Alexandru
  */
 public class HydraSettingsImpl extends JpsElementBase<HydraSettingsImpl> implements HydraSettings{
   public static final HydraSettings DEFAULT = new HydraSettingsImpl(new State());
 
-  private final State myState;
+  private final State state;
 
   public HydraSettingsImpl(State state) {
-    this.myState = state;
+    this.state = state;
   }
 
   @Override
-  public boolean isHydraEnabled() { return myState.isHydraEnabled; }
+  public boolean isHydraEnabled() { return state.isHydraEnabled; }
 
   @Override
   public String getHydraVersion() {
-    return myState.hydraVersion;
+    return state.hydraVersion;
   }
 
+  @Override
+  public String getNumberOfCores() {
+    return state.noOfCores;
+  }
+
+  @Override
+  public String getHydraStorePath() { return state.hydraStorePath; }
+
+  @Override
+  public String getSourcePartitioner() { return state.sourcePartitioner; }
+
+  @Override
+  public String getProjectRoot() { return state.projectRoot; }
```

Comment:
Can we have these as Path or because of serialization/deserialization we need to keep using String? (wonder what they do in the Scala support)